erpnext/payroll/doctype/payroll_entry/payroll_entry.py

```diff
@@ -68,14 +70,26 @@ def get_emp_list(self):
 					t1.name = t2.employee
 					and t2.docstatus = 1
 			%s order by t2.from_date desc
-			""" % cond, {"sal_struct": tuple(sal_struct), "from_date": self.end_date}, as_dict=True)
+			""" % cond, {"sal_struct": tuple(sal_struct), "from_date": self.end_date, "payroll_payable_account": self.payroll_payable_account}, as_dict=True)
 			return emp_list
 
 	def fill_employee_details(self):
 		self.set('employees', [])
 		employees = self.get_emp_list()
 		if not employees:
-			frappe.throw(_("No employees for the mentioned criteria"))
+			error_msg = _("No employees found for the mentioned criteria - Company: {0}, Currency: {1}, Payroll Payable Account: {2}").format(
+				frappe.bold(self.company), frappe.bold(self.currency), frappe.bold(self.payroll_payable_account))
+			if self.branch:
+				error_msg += _(", Branch: {0}").format(frappe.bold(self.branch))
+			if self.department:
+				error_msg += _(", Department: {0}").format(frappe.bold(self.department))
+			if self.designation:
+				error_msg += _(", Designation: {0}").format(frappe.bold(self.designation))
+			if self.start_date:
+				error_msg += _(", Start date: {0}").format(frappe.bold(self.start_date))
+			if self.end_date:
+				error_msg += _(", End date: {0}").format(frappe.bold(self.end_date))
+			frappe.throw(error_msg, title=_("No employees found"))
```

Comment:
The message will be more readable like this:

"""
No employees found with the following selected criteria:
Company: 
Brnach:
Department:
"""